vars/piperExecuteBin.groovy

```diff
@@ -8,9 +8,14 @@ import com.sap.piper.Utils
 
 import static com.sap.piper.Prerequisites.checkScript
 
-void call(Map parameters = [:], stepName, metadataFile, List credentialInfo, failOnMissingReports = false, failOnMissingLinks = false) {
+void call(Map parameters = [:], stepName, metadataFile, List credentialInfo, failOnError = false, failOnMissingReports = false, failOnMissingLinks = false) {
```

Comment:
Woulnd't it be better to fail by default and (`failOnError = true`) and to have to switch off it explicitly?

Reply:
@OliverNocon the only thing which I don't find too nice is that we yet introduce another option for kind of "default mandatory steps" right? The fact that a pipeline can never tolerate errors in versioning and build but potentially in a security step like checkmarx is what we used to define via https://github.com/SAP/jenkins-library/blob/6eb8dc81914f55e28c4256b78dea36f16066810a/vars/handlePipelineStepErrors.groovy#L82.

Reply:
it's using the defaults, thus we should only override it if needed.

That is exactly what has been done earlier but did not make it into `golang` so far, see e.g. in
dockerExecute:
```
handlePipelineStepErrors(stepName: STEP_NAME, stepParameters: parameters, failOnError: true)
```

@nevskrem taking your feedback into consideration I just switched the parameters.
The `failOnError` is only required in very specific cases and as such go rather last ...

Reply:
@OliverNocon thanks! That sounds very reasonable and avoids continuous re-declaration.